src/components/side-panel/side-panel.component.ts

```diff
@@ -72,17 +78,25 @@ export class SidePanelComponent implements OnInit, OnDestroy {
     }
 
     get cssWidth(): string {
-        if (typeof this.width === 'number') {
-            return this.width === 0 ? '0' : this.width + 'px';
-        }
-        return this.width;
+        return this.getCssValue(this.width);
     }
 
     get cssTop(): string {
-        if (typeof this.top === 'number') {
-            return this.top === 0 ? '0' : this.top + 'px';
+        return this.getCssValue(this.top);
+    }
+
+    get cssMinWidth(): string {
+        if (!this.minWidth || this.minWidth === 'none') {
+            return 'none';
         }
-        return this.top;
+        return this.getCssValue(this.minWidth);
+    }
+
+    get cssMaxWidth(): string {
+        if (!this.maxWidth || this.maxWidth === 'none') {
+            return 'none';
```

Comment:
This would convert 0 to none, which isn't accurate. Just pass the value through if it isn't a number, it's documented as a CSS value so no need to overcomplicate it.